services-validation.md

```diff
@@ -557,53 +557,139 @@ Then in your call to `Validator::make` use the `Lang:get` to use your custom fil
+There are a variety of helpful validation rules; however, you may wish to specify some of your own.
 
-    Validator::extend('foo', function($attribute, $value, $parameters) {
-        return $value == 'foo';
-    });
+The recommended way of adding your own validation rule is to extend the Validator instance via the `extend` method. In an October CMS plugin, this can be added to the `boot()` callback method inside your `Plugin.php` registration file.
+
+You can extend the Validator instance with your custom validation rule as a `Closure`, or as a `Rule` object.
+
+#### Using Closures
+
+If you only need the functionality of a custom rule specified once throughout your plugin or application, you may use a Closure to define the rule. The first parameter defines the name of your rule, and the second parameter provides your Closure.
+
+    use Validator;
+
+    public function boot()
+    {
+        Validator::extend('foo', function($attribute, $value, $parameters) {
+            return $value == 'foo';
+        });
+    }
 
 The custom validator Closure receives three arguments: the name of the `$attribute` being validated, the `$value` of the attribute, and an array of `$parameters` passed to the rule.
 
 You may also pass a class and method to the `extend` method instead of a Closure:
 
     Validator::extend('foo', 'FooValidator@validate');
 
-Note that you will also need to define an error message for your custom rules. You can do so either using an inline custom message array or by adding an entry in the validation language file.
+Once the Validator has been extended with your custom rule, you will need to add it to your rules definition. For example, you may add it to the `$rules` array of your model.
+
+    public $rules = [
+        'field' => 'foo'
+    ];
 
-#### Extending the validator class
+#### Using Rule objects
 
-Instead of using Closure callbacks to extend the Validator, you may also extend the Validator class itself. To do so, write a Validator class that extends `Illuminate\Validation\Validator`. You may add validation methods to the class by prefixing them with `validate`:
+A `Rule` object represents a single reusable validation rule for your models that implements the `Illuminate\Contracts\Validation\Rule` contract. Each rule object must provide three methods: a `passes` method which determines if a given value passes validation, a `validate` method which is called on validation and a `message` method which defines the default fallback error message.
 
     <?php
+    use Illuminate\Contracts\Validation\Rule;
 
-    class CustomValidator extends Illuminate\Validation\Validator
+    class Uppercase implements Rule
     {
+        /**
+        * Determine if the validation rule passes.
+        *
+        * @param  string  $attribute
+        * @param  mixed  $value
+        * @return bool
+        */
```

Comment:
Docblocks should have one space before every `*` in every line but the first I believe. This is a fantastic addition to the documentation though!

Reply:
You're correct. I don't believe the API generators care either way, but the [PSR-5 draft](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc.md#4-basic-principles) makes it explicit. TIL :)